### PR TITLE
ocamlPackages.conduit: 1.0.0 -> 1.4.0 and dependencies

### PR DESCRIPTION
**Pull request by alexfmpe**
###### Motivation for this change
https://github.com/mirage/ocaml-conduit/blob/master/CHANGES.md#v140
https://github.com/mirage/ocaml-ipaddr/blob/master/CHANGES.md#v310-2019-03-02

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/ocaml-modules/conduit/default.nix [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-		description = "Network connection library for TCP and SSL";
-		license = stdenv.lib.licenses.isc;
-		maintainers = [ stdenv.lib.maintainers.vbgl ];
-		inherit (src.meta) homepage;
-	};
+  meta = {
+    description = "Network connection library for TCP and SSL";
+    license = stdenv.lib.licenses.isc;
+    maintainers = with stdenv.lib.maintainers; [ alexfmpe vbgl ];
+    inherit (ocaml.meta) platforms;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/conduit/default.nix]: This line is not needed: `buildDunePackage` sets `meta.platforms` to this value.

File: pkgs/development/ocaml-modules/conduit/lwt.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, buildDunePackage, ppx_sexp_conv, conduit, ocaml_lwt }:
+{ stdenv, buildDunePackage, ppx_sexp_conv, sexplib, conduit, ocaml_lwt }:
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/conduit/lwt.nix]: This change is dubious.

File: pkgs/development/ocaml-modules/ipaddr/default.nix [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
+  buildInputs = [ macaddr ounit sexplib ];
+
+  doCheck = true;
 
   meta = with stdenv.lib; {
     homepage = https://github.com/mirage/ocaml-ipaddr;
     description = "A library for manipulation of IP (and MAC) address representations ";
     license = licenses.isc;
-    maintainers = [ maintainers.ericbmerritt ];
+    inherit (ocaml.meta) platforms;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ipaddr/default.nix]: This line is not needed.

File: pkgs/development/ocaml-modules/macaddr/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  propagatedBuildInputs = [ ppx_sexp_conv ];
+
+  doCheck = false; # ipaddr and macaddr tests are together, which requires mutual dependency
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/mirage/ocaml-ipaddr;
+    description = "A library for manipulation of MAC address representations";
+    license = licenses.isc;
+    inherit (ocaml.meta) platforms;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/macaddr/default.nix]: Not needed.

**Review by vbgl — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/conduit/lwt.nix (hunk 2 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/conduit/lwt.nix]: Huh indeed - must be some leftovers from intermediate state

File: pkgs/development/ocaml-modules/conduit/default.nix (hunk 1 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/conduit/default.nix]: Ah thanks - removed it

File: pkgs/development/ocaml-modules/ipaddr/default.nix [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
+{ stdenv, fetchurl, buildDunePackage, ocaml
+, macaddr, ounit, sexplib
+}:
 
 buildDunePackage rec {
   pname = "ipaddr";
-  version = "2.8.0";
+  version = "3.1.0";
 
   src = fetchurl {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ipaddr/default.nix]: The `version` and `src` attributes should be inherited from `macaddr`.

File: pkgs/development/ocaml-modules/macaddr/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  minimumOCamlVersion = "4.04";
+
+  src = fetchFromGitHub {
+    owner = "mirage";
+    repo = "ocaml-ipaddr";
+    rev = "v${version}";
+    sha256 = "0g7wgfn06z7fhdk4hlblf7b5k2ws56z0l73gk7189p5wvrmbavvx";
+  };
+
+  buildInputs = [ sexplib ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/macaddr/default.nix]: Why is `sexplib` needed?

File: pkgs/development/ocaml-modules/ipaddr/default.nix [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
-    url = "https://github.com/mirage/ocaml-${pname}/archive/${version}.tar.gz";
-    sha256 = "1amb1pbm9ybpxy6190qygpj6nmbzzs2r6vx4xh5r6v89szx9rfxw";
+    url = "https://github.com/mirage/ocaml-${pname}/archive/v${version}.tar.gz";
+    sha256 = "1hi3v5dzg6h4qb268ch3h6v61gsc8bv21ajhb35z37v5nsdmyzbh";
   };
 
-  propagatedBuildInputs = [ ppx_sexp_conv sexplib ];
+  buildInputs = [ macaddr ounit sexplib ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ipaddr/default.nix]: `macaddr` should be propagated.

Comment (vbgl) [inline pkgs/development/ocaml-modules/ipaddr/default.nix]: Why is `sexplib` needed?

File: pkgs/development/ocaml-modules/conduit/default.nix [hunk 8]
```diff
@@ -1,26 +1,26 @@
-{ stdenv, fetchFromGitHub, buildDunePackage
-, ppx_sexp_conv
-, astring, ipaddr, uri
+{ stdenv, fetchFromGitHub, buildDunePackage, ocaml
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/conduit/default.nix]: The `ocaml` parameter is never used.

**Review by vbgl — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/ipaddr/default.nix (hunk 5 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/ipaddr/default.nix]: Ah yes, makes sense.

File: pkgs/development/ocaml-modules/macaddr/default.nix (hunk 6 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/macaddr/default.nix]: Good catch, looks like it isn't - I don't quite remember what made me add it, but it was probably a misguided attempt to fix the following error, which I get if I `ppx_sexp_conv` isn't a dependency  
```
building '/nix/store/vm9lr1n0cj93iqc0xx5rl3plxlwnw9qp-ocaml4.07.1-macaddr-3.1.0.drv'...
unpacking sources
unpacking source archive /nix/store/zljnp91y7ig1av99d3l0kiacpn39b3ab-v3.1.0.tar.gz
source root is ocaml-ipaddr-3.1.0
setting SOURCE_DATE_EPOCH to timestamp 1551633159 of file ocaml-ipaddr-3.1.0/macaddr.opam
patching sources
configuring
no configure script, doing nothing
building
File "lib/dune", line 22, characters 20-28:
22 |  (libraries macaddr sexplib0))
                         ^^^^^^^^
Error: Library "sexplib0" not found.
```

File: pkgs/development/ocaml-modules/ipaddr/default.nix (hunk 7 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/ipaddr/default.nix]: It actually isn't - see the reply to the same question in `macaddr`.

Comment (alexfmpe) [reply pkgs/development/ocaml-modules/ipaddr/default.nix]: Done

File: pkgs/development/ocaml-modules/conduit/default.nix (hunk 8 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/conduit/default.nix]: Oops.

File: pkgs/development/ocaml-modules/macaddr/default.nix (hunk 6 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/macaddr/default.nix]: The `sexplib0` library is packaged under the `sexplib0` attribute. It is different than `sexplib`.

File: pkgs/development/ocaml-modules/ipaddr/default.nix [hunk 9]
```diff
@@ -1,20 +1,22 @@
-{ stdenv, fetchurl, buildDunePackage, sexplib, ppx_sexp_conv }:
+{ stdenv, fetchurl, buildDunePackage, ocaml
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ipaddr/default.nix]: The `fetchurl`, `ocaml` and `sexplib` arguments are not used.

File: pkgs/development/ocaml-modules/macaddr/default.nix [hunk 10]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl, buildDunePackage, ocaml
+, ppx_sexp_conv, sexplib
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/macaddr/default.nix]: The `ocaml` and `sexplib` arguments are not used.

**Review by vbgl — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/ipaddr/default.nix (hunk 9 above)
Comment (alexfmpe) [reply pkgs/development/ocaml-modules/ipaddr/default.nix]: Yikes, sorry to force several rounds of this on you. Thanks for staying on top of things.
I just found https://github.com/Synthetica9/nix-linter and used that to find any pesky leftovers.

**Review by vbgl — APPROVED** [review]

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.cohttp-lwt-unix

**Pull request merged**